tasks/object-detection-yolov8/src/predict.py

```diff
@@ -13,27 +13,46 @@ def classByLabelId(labelId: int, classes: ImageDatasetClasses) -> Optional[Image
     return classes.classByLabel(classes.labels[labelId])
 
 
-def run(model: YOLO, dataset: ComputerVisionDataset, resultPath: Path) -> None:
-    for sample in dataset.samples:
-        result: Results = model.predict(sample.imagePath, save = True, project = "./results")[0]
-        plt.imshow(result.orig_img)
+def processResult(result: Results, classes: list[ImageDatasetClasses], savePath: Path):
+    fig = plt.figure(num = 1, clear = True)
+    plt.imshow(result.orig_img)
 
-        if result.boxes is not None:
-            for minX, minY, maxX, maxY, confidence, labelId in result.boxes.data:
-                box = BBox.create(minX, minY, maxX, maxY)
+    if result.boxes is not None:
+        for minX, minY, maxX, maxY, confidence, labelId in result.boxes.data:
+            box = BBox.create(minX, minY, maxX, maxY)
 
-                clazz = classByLabelId(int(labelId), dataset.classes)
-                if clazz is None:
-                    continue
+            clazz = classByLabelId(int(labelId), classes)
+            if clazz is None:
+                continue
 
-                plt.gca().add_patch(pth.Rectangle(
-                    (float(box.minX), float(box.minY)),
-                    float(box.width),
-                    float(box.height),
-                    linewidth = 3,
-                    edgecolor = clazz.color,
-                    facecolor = "none"
-                ))
+            fig.gca().add_patch(pth.Rectangle(
+                (float(box.minX), float(box.minY)),
+                float(box.width),
+                float(box.height),
+                linewidth = 3,
+                edgecolor = clazz.color,
+                facecolor = "none"
+            ))
 
-        plt.savefig(resultPath / f"{sample.id}.png")
-        plt.close()
+    plt.savefig(savePath)
+
+
+def isSampleValid(sample: ComputerVisionSample) -> bool:
+    for instance in sample.load().annotation.instances:
+        if any([len(segmentation) < 6 for segmentation in instance.segmentations]):
+            return False
+
+    return True
+
+
+def run(model: YOLO, dataset: ComputerVisionDataset, resultPath: Path, batchSize: int) -> None:
+    for i in range(0, len(dataset.samples), batchSize):
+        print(i)
+        print(dataset.count - i)
+        print(i + batchSize)
+        if (dataset.count - i) < batchSize:
+            batchSize = (dataset.count - i)
```

Comment:
Why is there a need for this condition?

Reply:
To adjust the final batch to the size of the remainder

Reply:
Let's do processing of remainder outside of the for-loop. It is easy to get the reminder with modulo operator